src/backend/distributed/planner/distributed_planner.c

```diff
@@ -810,13 +817,43 @@ CreateDistributedPlan(uint64 planId, Query *originalQuery, Query *query, ParamLi
 	/* distributed plan currently should always succeed or error out */
 	Assert(distributedPlan && distributedPlan->planningError == NULL);
 
-	/* traverse distributedPlan and extract the list of used subplans */
-	distributedPlan->usedSubPlanNodeList = FindSubPlansUsedInPlan(distributedPlan);
+	RecordSubPlansUsedInPlan(distributedPlan, originalQuery);
 
 	return distributedPlan;
 }
 
 
+/*
+ * RecordSubPlansUsedInPlan gets a distributed plan a queryTree, and
+ * updates the usedSubPlanNodeList of the distributed plan.
+ *
+ * The function simply pulls all the subPlans that are used in the queryTree
+ * with one exception: subPlans in the HAVING clause. The reason is explained
+ * in the function.
+ */
+static void
+RecordSubPlansUsedInPlan(DistributedPlan *plan, Query *queryTree)
```

Comment:
maybe preserve `originalQuery` for consistency, since `queryTree` sometimes refers to the rewritten query.